File: Frontend/WebUI/ElectricGo/src/app/shared/createUserDTO.ts
export class ICreateUserDTO{
  userName: string;
  role: string;
  email: string;
  phoneNumber : string;
  birthDate : string;


  constructor(userName : string, role : string, email : string, phoneNumber : string, birthDate : string) {

    this.userName = userName;
    this.role = role;
    this.email = email;
    this.phoneNumber = phoneNumber;
    this.birthDate = birthDate;
  }
}

File: Backend/GestArm_API/GestArm/Domain/Users/CreateUserDTO.cs
namespace GestArm.Domain.Users;

public class CreateUserDTO
{
    public CreateUserDTO(string userName, string role, string email, string phoneNumber, string birthDate)
    {
        Name = userName;
        Role = role;
        Email = email;
        PhoneNumber = phoneNumber;
        Activated = true;
        BirthDate = birthDate;
    }

    public string Name { get; set; }
    public string Email { get; set; }
    public string Role { get; set; }
    public string PhoneNumber { get; set; }
    public string BirthDate { get; set; }
    public bool Activated { get; set; }
}